compiler/parser/python.lalrpop

```diff
@@ -416,10 +434,19 @@ TryStatement: ast::Stmt = {
     <location:@L> "try" ":" <body:Suite> <handlers:ExceptClause+> <else_suite:("else" ":" Suite)?> <finally:("finally" ":" Suite)?> <end_location:@R> => {
         let orelse = else_suite.map(|s| s.2).unwrap_or_default();
         let finalbody = finally.map(|s| s.2).unwrap_or_default();
+        let end_location = if let Some(last) = finalbody.last() {
+            last.end_location
+        } else {
+            if let Some(last) = orelse.last() {
+                last.end_location
+            } else {
+                handlers.last().unwrap().end_location
+            }
+        };
```

Comment:
Asked ChatGPT if we can simplify this code:

<img width="814" alt="image" src="https://user-images.githubusercontent.com/17039389/206940916-f11d3b9b-dfd3-4da4-991f-c05ff617eea0.png">